src/app/services/project.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+
+import { Project } from '../models/project';
+
+@Injectable()
+export class ProjectService {
+
+    private storageKey: string = 'cclab.projects';
+    private projects: Project[] = [];
+    private storage: Storage = window.localStorage;
```

Comment:
Yeah, it definitely should be injected. I attempted this, and it was more of a pain than I expected.

So I decided to defer, because this use of localStorage may not persist if we switch to having the API user account store the project information on the server.